runtimes/core/src/api/reqauth/platform.rs

```diff
@@ -69,6 +70,10 @@ impl RequestValidator {
     }
 
     pub fn sign_outgoing_request(&self, req: &mut reqwest::Request) -> anyhow::Result<()> {
+        let path = percent_decode_str(req.url().path())
+            .decode_utf8_lossy()
+            .to_string();
```

Comment:
I don't think we need `.to_string()` here? `Cow<'_, str>` should have `as_bytes()` too.

Reply:
This is to battle the borrow checker, `req` is mutually borrowed below when we insert the date header. So we can't also have a non-mutual reference at the same time